### PR TITLE
Fix Pending changes aren't updated when renaming the topic name

### LINKED ISSUE #4154: 🐛 Bug Report:  Pending changes aren't updated when renaming the topic name
### What happened?

 Pending changes aren't updated when renaming the topic name

https://user-images.githubusercontent.com/39680385/195817290-762fc124-74b7-4ec0-92cd-8e4285acd949.mov

### What you expected to happen

I expect that the topic name on the pending changes will be updated every time the user renames the topic name

### How to reproduce

see the video

### Amplication version

1.0.2

### Environment

_No response_

### Are you willing to submit PR?

_No response_

### PR DESCRIPTION
**Pull request by SayyidMuhammadA**
Close #4154

## PR Details
Fix Pending changes aren't updated when renaming the topic name

## PR Checklist
- [ ] Tests for the changes have been added
- [ ] `npm test` doesn't throw any error

### CODE REVIEW

File: packages/amplication-client/src/VersionControl/PendingChange.tsx [hunk 1]
```diff
@@ -1,34 +1,55 @@
...
+    variables: {
+      where: {
+        resource: { id: 'cl9kzud3f24405wul5cj4u55f' },
+        displayName: undefined,
+      },
+      orderBy: {
+        [DATE_CREATED_FIELD]: models.SortOrder.Asc,
+      },
+    },
+  });
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/VersionControl/PendingChange.tsx:51]: The PendingChange component should not know anything specific about Topics.
Also, please note that you used hard-coded value of resourceId

I am rejecting this change as it is. The fix should be implemented in the Topic 
see this as an example 
https://github.com/amplication/amplication/blob/ed80357733cc315eaa7a8da92f6d7f6313f9bce2/packages/amplication-client/src/Plugins/hooks/usePlugins.ts#L179

**Review by yuval-hazaz — CHANGES_REQUESTED** [review]

File: packages/amplication-client/src/VersionControl/PendingChange.tsx (hunk 1 above)
Comment (SayyidMuhammad878) [reply packages/amplication-client/src/VersionControl/PendingChange.tsx:51]: Do you mean I can change the resource id to ```data.setPluginOrder.id```

Comment (yuval-hazaz) [reply packages/amplication-client/src/VersionControl/PendingChange.tsx:51]: I mean that you need to use `AddBlock` in `Topic.tsx` and `NewTopic.tsx`

Comment (SayyidMuhammad878) [reply packages/amplication-client/src/VersionControl/PendingChange.tsx:51]: Okay, Thanks

Comment (souravjain540) [reply packages/amplication-client/src/VersionControl/PendingChange.tsx:51]: any updates? @SayyidMuhammad878

Comment (yuval-hazaz) [reply packages/amplication-client/src/VersionControl/PendingChange.tsx:51]: @SayyidMuhammad878  all good. We understand you may be busy. Let us know if you need us to take it over.

Comment (SayyidMuhammad878) [reply packages/amplication-client/src/VersionControl/PendingChange.tsx]: I've fixed it you can check it out

**Review by yuval-hazaz — CHANGES_REQUESTED** [review]
Thanks @SayyidMuhammad878 

unfortunately, we cannot accept this PR. 
As explained before, this PR should not include any changes in `pendingChange.tsx` or `pendingChanges.tsx` - these two component should not know and should not be aware of anything specific about Topics, or about any other type of changes. 

The soltion is that when creating or updating a topic the code in the original should call `AddBlock` method just like I referenced in the previous comment. 

I understand that in order to resolve this issue, you may need so previous knowledge about Amplication client and its architecture, so it is totally ok and understood. 

Thank you very much for putting your time and effort to resolve this issue. Your contribution is appreciated.

**Comment by yuval-hazaz** [comment]
I am closing this PR.
@SayyidMuhammadA  please feel free to pick up an issue that is available for the community when you are ready

**Pull request closed without merging**